### PR TITLE
issues#110 Prevent java.lang.ArrayIndexOutOfBoundsException

### PR DESCRIPTION
**Pull request by AlexanderLosev**
Prevent java.lang.ArrayIndexOutOfBoundsException on PagedLongStack.peek() when top node is in the previous page

### CODE REVIEW

**Comment by Mats-SX** [comment]
Hi @AlexanderLosev and thanks for your contribution! I'm sorry about the slow reply.

In order to merge your fix, there are a few more things that we'll need:

* First, we'll need you to sign the [Neo4j CLA (Contributor License Agreement)](https://neo4j.com/developer/cla/) so that we are allowed to ship your code in our library.
* Second, it would be great with a test case that breaks in case this code is modified in the future in a way that again provokes the error. It would also serve as evidence that there is a problem needing fixing here.

Normally, I would also request a small benchmark that shows that the code does not contribute to a performance regression, but from a quick peek (heh) I'm not sure it's necessary. Still, if you would like to do the due diligence it would be an extra gold star.

With these small modifications I would be happy to help merge your contribution!

All the best
Mats

File: core/src/test/java/org/neo4j/graphalgo/core/utils/paged/PagedLongStackTest.java [hunk 1]
```diff
@@ -82,6 +82,28 @@ void shouldPeekLastAddedValue() {
...
+            long value = between(0L, 1337L).Long();
+            expected = expectedLast;
+            expectedLast = value;
+            stack.push(value);
+        }
+
+        long last = stack.pop();
+        long actual = stack.peek();
+        final String reason = "Wrong Value at index " + stack.pageSize;
+        final String reasonLast = "Wrong Value at index " + repetitions;
```
Comment (Mats-SX) [inline core/src/test/java/org/neo4j/graphalgo/core/utils/paged/PagedLongStackTest.java:102]: could be `var`s

Comment (Mats-SX) [reply core/src/test/java/org/neo4j/graphalgo/core/utils/paged/PagedLongStackTest.java:102]: or even inlined

**Review by Mats-SX — APPROVED** [review]
Looks great!

**Comment by Mats-SX** [comment]
Hi @AlexanderLosev! I can confirm that we have received your CLA signature, and I'll be happy to merge this. I will merge this into our private repository to all branches and you should be seeing the result on this repository soon. At that point I will close this PR.

Thanks a lot for the contribution!

**Comment by FlorentinD** [comment]
As the PR is merged, I will close this PR

**Pull request closed without merging**